test-basic-proxy/lib/positive/sessionsTest.js

```diff
@@ -22,139 +22,142 @@ const testutil = require('../testutil');
+            })
+            .then(output => {
+                expect(output).to.equal(false);
+                return service.checkTransaction(transactionSession, docUri);
+            })
+            .then(output => {
+                expect(output).to.equal(false);
+                done();
+            })
+            .catch(err => {
+                if (!hasRolledBack) {
+                    service.rollbackTransaction(transactionSession);
+                }
+                expect.fail(err.toString());
+                done();
             });
     });
-    it('field without session', function(done) {
-      service.setSessionFieldNoSession('shouldNeverSucceed')
-          .then(output => {
-            expect.fail(err.toString());
-            done();
-            })
-          .catch(err => {
-            expect(err.toString()).to.equal(
-                'Error: call to /dbf/test/sessions/setSessionFieldNoSession.sjs: cannot process response with 500 status'
-            );
-            done();
+    it('concurrent', function(done) {
+        this.timeout(3000);
+        const concurrentMax = 3;
+        const sleeptime = 750;
+        let running = concurrentMax;
+        for (let i=0; i < concurrentMax; i++) {
+            const session = service.createSession();
+            service.sleepify(session, sleeptime)
+                .then(output => {
+                    expect(output).to.equal(true);
+                    if (--running === 0) {
+                        done();
+                    }})
+                .catch(err => {
+                    expect.fail(err.toString());
+                    done();
+                });
+        }
+    });
+    describe('negative', function() {
+        it('null', function(done) {
+            try {
+                service.setSessionField(null, 'unused')
+                    .then(output => {
+                        expect.fail(err.toString());
+                        done();
+                    })
+                    .catch(err => {
+                        expect.fail(err.toString());
+                        done();
+                    });
+                expect.fail('error not thrown');
+            } catch(err) {
+                expect(err.toString()).to.equal('Error: missing required session parameter: timestamper');
+                done();
+            }
+        });
+        it('transaction without session', function(done) {
+            service.beginTransactionNoSession('/test/session/transaction/negative1.txt', 'should never succeed')
+                .then(output => {
+                    done();
+                })
+                .catch(err => {
+                    expect(err.toString()).to.equal(
+                        'Error: call to /dbf/test/sessions/beginTransactionNoSession.sjs: cannot process response with 500 status'
+                    );
+                    done();
+                }).finally(() => {
+                done();
```

Comment:
@mariusmarklogic instead of calling done() in finally block, we need to investigate why it is failing. I recommend adding a try-catch block around line 140 and see if the failure is there. Same for line line 158

Reply:
I put the finally block because the error seemed to be the exact same thing when i looked for the first time, but now I observed that it seems to be from Operation.js(227), which includes "with path: " after the error. I changed it assertion from "equal" to "contain".